packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/plugin-ledger-connector-iroha.ts

```diff
@@ -0,0 +1,684 @@
+
+export interface IPluginLedgerConnectorIrohaOptions
+  extends ICactusPluginOptions {
+  rpcToriiPortHost: string;
+  pluginRegistry: PluginRegistry;
+  prometheusExporter?: PrometheusExporter;
+  logLevel?: LogLevelDesc;
+}
+
+export class PluginLedgerConnectorIroha
+  implements
+    IPluginLedgerConnector<
+      never,
+      never,
+      RunTransactionRequest,
+      RunTransactionResponse
+    >,
+    ICactusPlugin,
+    IPluginWebService {
+  private readonly instanceId: string;
+  public prometheusExporter: PrometheusExporter;
+  private readonly log: Logger;
+  private readonly pluginRegistry: PluginRegistry;
+
+  private endpoints: IWebServiceEndpoint[] | undefined;
+  private httpServer: Server | SecureServer | null = null;
+
+  public static readonly CLASS_NAME = "PluginLedgerConnectorIroha";
+
+  public get className(): string {
+    return PluginLedgerConnectorIroha.CLASS_NAME;
+  }
+
+  constructor(public readonly options: IPluginLedgerConnectorIrohaOptions) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+    Checks.truthy(
+      options.rpcToriiPortHost,
+      `${fnTag} options.rpcToriiPortHost`,
+    );
+    Checks.truthy(options.pluginRegistry, `${fnTag} options.pluginRegistry`);
+    Checks.truthy(options.instanceId, `${fnTag} options.instanceId`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = options.instanceId;
+    this.pluginRegistry = options.pluginRegistry;
+    this.prometheusExporter =
+      options.prometheusExporter ||
+      new PrometheusExporter({ pollingIntervalInMin: 1 });
+    Checks.truthy(
+      this.prometheusExporter,
+      `${fnTag} options.prometheusExporter`,
+    );
+
+    this.prometheusExporter.startMetricsCollection();
+  }
+  deployContract(): Promise<never> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getPrometheusExporter(): PrometheusExporter {
+    return this.prometheusExporter;
+  }
+
+  public async getPrometheusExporterMetrics(): Promise<string> {
+    const res: string = await this.prometheusExporter.getPrometheusMetrics();
+    this.log.debug(`getPrometheusExporterMetrics() response: %o`, res);
+    return res;
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public async onPluginInit(): Promise<unknown> {
+    return;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.ofNullable(this.httpServer);
+  }
+
+  public async shutdown(): Promise<void> {
+    const serverMaybe = this.getHttpServer();
+    if (serverMaybe.isPresent()) {
+      const server = serverMaybe.get();
+      await promisify(server.close.bind(server))();
+    }
+  }
+
+  async registerWebServices(
+    app: Express,
+    //wsApi: SocketIoServer,
+  ): Promise<IWebServiceEndpoint[]> {
+    //const { web3 } = this;
```

Comment:
@hxlaf Please delete the commented lines unless there's a specific reason to keep them around.

Reply:
Fixed! Thank you!